tracer/src/Datadog.Trace/Debugger/ProbeStatuses/ProbeStatusPoller.cs

```diff
@@ -106,7 +120,7 @@ private void ResumePollerTimer()
 
         private void OnProbeStatusesPoll()
         {
-            if (!_probes.Any())
+            if (!_probes.Any() || _isDisposed)
```

Comment:
This _probes access is not protected by the lock and a concurrent call could potentially cause issues